### PR TITLE
Add set permission for certain packages

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

File: src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs [hunk 1]
```diff
@@ -218,23 +265,21 @@ private IEnumerable<PackageSource> LoadNuGetSources(PackageSourceLocation packag
...
             }
 
             if (!includePreview)
             {
-                (PackageSource, IPackageSearchMetadata) latestStableVersion = accumulativeSearchResults
-                    .Where(r => !r.package.Identity.Version.IsPrerelease)
-                    .DefaultIfEmpty(default)
-                    .MaxBy(r => r.package.Identity.Version);
+                var stableVersions = accumulativeSearchResults
+                    .Where(r => !r.package.Identity.Version.IsPrerelease);
```
Comment (wli3) [inline src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs:278]: This is a bug just discovered when accumulativeSearchResults is empty

File: src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs [hunk 2]
```diff
@@ -107,12 +110,56 @@ public async Task<IEnumerable<string>> ExtractPackageAsync(string packagePath, s
...
+            }
+
+            return files
+                .Where(FileUnderToolsWithoutSuffix)
+                .Select(f => new FilePath(f));
+        }
+
+        private static bool PackageIsInAllowList(IEnumerable<string> files)
+        {
+            var allowListOfPackage = new string[] {"microsoft.android.sdk.darwin"};
```
Comment (jonathanpeppers) [inline src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs:152]: @rolfbjarne can confirm tomorrow, but it looks like `Microsoft.iOS.Sdk` also has a `tools` folder where any file without an extension would need execute permission.

Rolf, are there any other files you know of?

Comment (rolfbjarne) [reply src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs:152]: These are our executable files:



### Microsoft.MacCatalyst.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.MacCatalyst.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.MacCatalyst.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.MacCatalyst.Sdk/tools/lib/bgen/bgen*

### Microsoft.iOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.iOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.iOS.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.iOS.Sdk/tools/lib/bgen/bgen*
-rwxr-xr-x  1 rolf  wheel  6037104 Apr 29 10:41:27 2021 Microsoft.iOS.Sdk/tools/lib/mlaunch/mlaunch.app/Contents/MacOS/mlaunch*

### Microsoft.macOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.macOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.macOS.Sdk/tools/lib/bgen/bgen*

### Microsoft.tvOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.tvOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.tvOS.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.tvOS.Sdk/tools/lib/bgen/bgen*
-rwxr-xr-x  1 rolf  wheel  6037104 Apr 29 10:41:27 2021 Microsoft.tvOS.Sdk/tools/lib/mlaunch/mlaunch.app/Contents/MacOS/mlaunch*

File: src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs [hunk 3]
```diff
@@ -107,12 +110,56 @@ public async Task<IEnumerable<string>> ExtractPackageAsync(string packagePath, s
...
+                }
+            }
+
+            return allFilesInPackage;
+        }
+
+        internal IEnumerable<FilePath> FindAllFilesNeedExecutablePermission(IEnumerable<string> files,
+            string targetPath)
+        {
+            if (!PackageIsInAllowList(files))
```
Comment (sfoslund) [inline src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs:134]: Is this a permanent solution? It seems like the allow list will get quickly out of hand when there are more workloads, are we planning on implementing a way for workloads to provide their own allow list?

**Review by sfoslund — APPROVED** [review]

File: src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs (hunk 2 above)
Comment (wli3) [reply src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs:152]: So I'll add Microsoft.MacCatalyst.Sdk, Microsoft.iOS.Sdk, Microsoft.macOS.Sdk, Microsoft.tvOS.Sdk

and just mark everything without an extension under tools executable

**Pull request merged**